File: app/src/main/java/com/dubedivine/samples/data/model/Media.java
package com.dubedivine.samples.data.model;




import java.io.Serializable;
import java.util.Date;

/**
 * Created by divine on 2017/08/13.
 */
public class Media implements Serializable {
    private String name;
    private long size;
    private char type;  // P-> picture, V-> video, f -> docs
    private String location;
    private Date createAt = new Date();

    public String getName() {
        return name;
    }

    public long getSize() {
        return size;
    }

    public char getType() {
        return type;
    }

    public String getLocation() {
        return location;
    }

    public int getLimit() {
        return 5120;
    }

    public Date getCreateAt() {
        return createAt;
    }

    public Media(String name, long size, char type, String location) {
        this.name = name;
        this.size = size;
        this.type = type;
        this.location = location;
    }
}

File: app/src/main/java/com/dubedivine/samples/data/model/Media.kt
package com.dubedivine.samples.data.model


import java.io.Serializable
import java.util.*

/**
 * Created by divine on 2017/08/13.
 */
// P-> picture, V-> video, f -> docs
class Media(val name: String,
            val size: Long,
            val type: Char,
            var location: String) : Serializable {

    val limit: Int
        get() = 5120

    companion object {
        //enums are very expensive so we use chars instead!!
        const val VIDEO_TYPE = 'V'
        const val PICTURE_TYPE = 'P'
        const val DOCS_TYPE = 'D'
    }
}
